test/common/stats/stat_test_utility.cc

```diff
@@ -12,7 +12,10 @@ bool hasDeterministicMallocStats() {
   // library for Envoy is TCMALLOC that's what we test for here. If we switch
   // to a different malloc library than we'd have to re-evaluate all the
   // thresholds in the tests referencing hasDeterministicMallocStats().
-#if defined(TCMALLOC) && !defined(ENVOY_MEMORY_DEBUG_ENABLED)
+  // TODO(cmluciano) Enable LIBCPP testing once stabilized
+#if _LIBCPP_VERSION
```

Comment:
in stats_integration_test.cc you used `#ifdef` rather than `#if` for that variable. Usually for `#if` I expect the preprocessor directive to be unconditionally defined to 0 or something else.